Sub-folders icons doesn't work unless the dev server restarted

### What version of `astro-icon` are you using?

"^1.0.2"

### Astro Info

```block
> astro "info"

Astro                    v4.3.1
Node                     v20.9.0
System                   Linux (x64)
Package Manager          pnpm
Output                   server
Adapter                  none
Integrations             @astrojs/react
                         astro-icon
```


### If this issue only occurs in one browser, which browser is a problem?

_No response_

### Describe the Bug

I followed the guide from the docs 
Here:
https://www.astroicon.dev/getting-started/

to organize my icons I add them in a subfolder, but once I add a new SVG file, It's not detected by the Astro path as the module is not found the only solution is to restart the dev server 

Note: I use inline SVG, I paste the code in .SVG file and use it like this 

`import { Icon } from 'astro-icon/components'
`
`<Icon name='product/CartSvg' width={24} height={24} />
`
The Folder looks like this 
src/icons/**/*
![image](https://github.com/natemoo-re/astro-icon/assets/37639473/1cf22134-c1da-4e18-82f1-d037676a1121)

Error: 02:04:58 
`[ERROR] Unable to locate "product/CartSvg" icon!`

### What's the expected result?

As I understand it should be detected and can be used without the need to restart every time I add a new SVG Icon.

### Link to Minimal Reproducible Example

https://github.com/FairyPenguin/astro-reproduce

--- a/.changeset/spicy-plums-hunt.md
+++ b/.changeset/spicy-plums-hunt.md
@@ -1,0 +1,5 @@
+---
+"astro-icon": patch
+---
+
+Fix icon dir watcher so newly added local icons (including files inside subfolders) are picked up without restarting the dev server

--- a/packages/core/src/vite-plugin-astro-icon.ts
+++ b/packages/core/src/vite-plugin-astro-icon.ts
@@ -49,15 +49,6 @@ export function createPlugin(
       }
     },
     configureServer({ watcher, moduleGraph }) {
-      // Vite's dev server watcher is created with `disableGlobbing: true`,
-      // so a glob string like `${iconDir}/**/*.svg` is treated as a literal
-      // (non-existent) path rather than a glob pattern. Chokidar's internal
-      // fallback logic for that missing path ends up re-adding `iconDir`
-      // with a broken, non-recursive filter, which silently stops emitting
-      // `add`/`change`/`unlink` events for the whole directory (including
-      // existing and new subfolders) until the dev server is restarted.
-      // Watching the resolved directory itself avoids that failure mode and
-      // ensures `iconDir` is watched even when it lives outside `root`.
       watcher.add(resolve(root.pathname, iconDir));
       watcher.on("all", async (_, filepath: string) => {
         const parsedPath = parse(filepath);

--- a/packages/core/src/vite-plugin-astro-icon.ts
+++ b/packages/core/src/vite-plugin-astro-icon.ts
@@ -49,7 +49,16 @@ export function createPlugin(
       }
     },
     configureServer({ watcher, moduleGraph }) {
-      watcher.add(`${iconDir}/**/*.svg`);
+      // Vite's dev server watcher is created with `disableGlobbing: true`,
+      // so a glob string like `${iconDir}/**/*.svg` is treated as a literal
+      // (non-existent) path rather than a glob pattern. Chokidar's internal
+      // fallback logic for that missing path ends up re-adding `iconDir`
+      // with a broken, non-recursive filter, which silently stops emitting
+      // `add`/`change`/`unlink` events for the whole directory (including
+      // existing and new subfolders) until the dev server is restarted.
+      // Watching the resolved directory itself avoids that failure mode and
+      // ensures `iconDir` is watched even when it lives outside `root`.
+      watcher.add(resolve(root.pathname, iconDir));
       watcher.on("all", async (_, filepath: string) => {
         const parsedPath = parse(filepath);
         const resolvedIconDir = resolve(root.pathname, iconDir);

